recipes/magnum/all/conanfile.py

```diff
@@ -0,0 +1,723 @@
+        "mesh_tools": True,
+        "primitives": True,
+        "scene_graph": True,
+        "shaders": True,
+        "text": True,
+        "texture_tools": True,
+        "trade": True,
+        "vk": True,
+
+        "android_application": True,
+        "emscripten_application": True,
+        "glfw_application": True,
+        "glx_application": True,
+        "sdl2_application": True,
+        "xegl_application": True,
+        "windowless_cgl_application": True,
+        "windowless_egl_application": True,
+        "windowless_glx_application": True,
+        "windowless_ios_application": True,
+        "windowless_wgl_application": True,
+        "windowless_windows_egl_application": True,
+
+        "cgl_context": True,
+        "egl_context": True,
+        "glx_context": True,
+        "wgl_context": True,
+
+        "gl_info": True,
+        "al_info": True,
+        "distance_field_converter": True,
+        "font_converter": True,
+        "image_converter": True,
+        "scene_converter": True,
+
+        # Related to plugins
+        "any_audio_importer": True,
+        "any_image_converter": True,
+        "any_image_importer": True,
+        "any_scene_converter": True,
+        "any_scene_importer": True,
+        "magnum_font": True,
+        "magnum_font_converter": True,
+        "obj_importer": True,
+        "tga_importer": True,
+        "tga_image_converter": True,
+        "wav_audio_importer": True,
+    }
+    generators = "cmake", "cmake_find_package"
+    exports_sources = ["CMakeLists.txt", "cmake/*"]
+    
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def config_options(self):
+        # Doc says that 'distance_field_converter' is only available with "desktop GL" (the same is said for 'font_converter', but it builds)
+        # TODO: Here we probably have a CHOICE OPTION
+        if self.options.target_gl in ["gles2", "gles3"]:
+            del self.options.distance_field_converter
+
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            del self.options.egl_context
+            del self.options.xegl_application
+            del self.options.windowless_egl_application
+            del self.options.windowless_ios_application
+            del self.options.windowless_glx_application
+            del self.options.windowless_windows_egl_application  # requires ANGLE
+            del self.options.target_headless  # Requires EGL (when used gl_info)
+            del self.options.glx_application
+            del self.options.cgl_context
+            del self.options.windowless_cgl_application
+
+        if self.settings.os == "Linux":
+            del self.options.cgl_context
+            del self.options.windowless_cgl_application
+            del self.options.wgl_context
```

Comment:
I think `wgl_context` (and every option containing `wgl`) is Windows-only:
https://github.com/mosra/magnum/blob/03b7adf71b1b50d2d1d2064b4a185355ee19dd5f/CMakeLists.txt#L167

Reply:
I can build `wgl_context` in Macos. TBH, I have no idea if it is usable or not. I disabled only the options that were failing for each OS, I find it very hard to reproduce the logic encoded in the build system.